incubator/hnc/internal/kubectl/describe.go

```diff
@@ -67,34 +74,72 @@ var describeCmd = &cobra.Command{
 			fmt.Printf("  No children\n")
 		}
 
-		// Early exit if no conditions
+		// Conditions
 		if len(hier.Status.Conditions) == 0 {
 			fmt.Printf("  No conditions\n")
-			return
-		}
-
-		// Conditions
-		fmt.Printf("  Conditions:\n")
-		for _, c := range hier.Status.Conditions {
-			fmt.Printf("  - %s: %s\n", c.Code, c.Msg)
-			if len(c.Affects) == 0 {
-				continue
-			}
-			fmt.Printf("    Affected by this condition:\n")
-			for _, a := range c.Affects {
-				if a.Name != "" {
-					if a.Group == "" {
-						a.Group = "core"
+		} else {
+			fmt.Printf("  Conditions:\n")
+			for _, c := range hier.Status.Conditions {
+				fmt.Printf("  - %s: %s\n", c.Code, c.Msg)
+				if len(c.Affects) == 0 {
+					continue
+				}
+				fmt.Printf("    Affected by this condition:\n")
+				for _, a := range c.Affects {
+					if a.Name != "" {
+						if a.Group == "" {
+							a.Group = "core"
+						}
+						fmt.Printf("      - %s/%s (%s/%s/%s)\n", a.Namespace, a.Name, a.Group, a.Version, a.Kind)
+					} else {
+						fmt.Printf("      - %s\n", a.Namespace)
 					}
-					fmt.Printf("      - %s/%s (%s/%s/%s)\n", a.Namespace, a.Name, a.Group, a.Version, a.Kind)
-				} else {
-					fmt.Printf("      - %s\n", a.Namespace)
 				}
 			}
 		}
+
+		// Events
+		describeEvents(nnm)
 	},
 }
 
+func describeEvents(nnm string) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	events, _ := k8sClient.CoreV1().Events(nnm).List(ctx, metav1.ListOptions{})
+	// filter out HNC events only
+	hncEvents := []v1.Event{}
+	for _, event := range events.Items {
+		if event.Source.Component == "hnc.x-k8s.io" {
+			hncEvents = append(hncEvents, event)
+		}
+	}
+	if len(hncEvents) == 0 {
+		fmt.Printf("\nNo recent HNC events for objects in this namespace\n")
+	} else {
```

Comment:
This is a great place for an early return so that you don't need the rest of the function in an `else` block. That's another great reason to use functions to organize code.

Reply:
Done